examples/src/main/java/com/mapbox/navigation/examples/NavigationSettingsActivity.java

```diff
@@ -51,8 +59,29 @@ protected boolean isValidFragment(String fragmentName) {
     public void onCreate(final Bundle savedInstanceState) {
       super.onCreate(savedInstanceState);
       addPreferencesFromResource(R.xml.fragment_navigation_preferences);
+
       String gitHashTitle = String.format("Last Commit Hash: %s", BuildConfig.GIT_HASH);
       findPreference(getString(R.string.git_hash_key)).setTitle(gitHashTitle);
+
+      findPreference(getString(R.string.nav_native_history_retrieve_key)).setOnPreferenceClickListener(preference -> {
+        String history = MapboxNativeNavigatorImpl.INSTANCE.getHistory();
```

Comment:
We can check for `EMPTY_HISTORY` to prevent writing empty files 👀 https://github.com/mapbox/mapbox-navigation-android/blob/d41f4bbda15a8c82d6b439f6a1482502f9690b6c/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/StoreHistoryTask.java#L29